### PR TITLE
fix: processing mls-welcome-message [WPB-7216]

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-7216" title="WPB-7216" target="_blank">BugWPB-7216</a>  [iOS] Missing messages in group conversation
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When the app is in background, new incoming MLS conversations won't be joined properly and messages are lost, until the app becomes active.

### Causes (Optional)

Failing to handle mls-welcome-message.

### Solutions

Restructure code and make sure, that joining an MLS conversation works in each case.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-request-strategy/Tests/Helpers/MessagingTestBase.swift [hunk 1]
```diff
@@ -560,4 +560,23 @@ extension MessagingTestBase {
...
+                                  fromID: UUID) -> NSMutableDictionary? {
+
+        return ["conversation": conversationID.transportString(),
+                "qualified_conversation": [
+                    "id": conversationID.transportString(),
+                    "domain": domain
+                ],
+                "data": data,
+                "from": fromID.transportString(),
+                "time": time?.transportString() ?? "",
```
Comment (netbe) [inline wire-ios-request-strategy/Tests/Helpers/MessagingTestBase.swift:577]: does backend allow empty string here?

**Review by netbe — COMMENTED** [review]
LGTM;) just a question

**Review by netbe — APPROVED** [review]

File: wire-ios-request-strategy/Tests/Helpers/MessagingTestBase.swift (hunk 1 above)
Comment (typfel) [reply wire-ios-request-strategy/Tests/Helpers/MessagingTestBase.swift:577]: No probably not

Comment (netbe) [reply wire-ios-request-strategy/Tests/Helpers/MessagingTestBase.swift:577]: ah but this is for tests

**Review by markusfassbender — APPROVED** [review]

File: wire-ios-data-model/Source/MLS/MLSService.swift [hunk 2]
```diff
@@ -816,36 +796,7 @@ public final class MLSService: MLSServiceInterface {
...
-            if let newDistributionPoints = CRLsDistributionPoints(
-                from: welcomeBundle.crlNewDistributionPoints
-            ) {
-                onNewCRLsDistributionPointsSubject.send(newDistributionPoints)
-            }
-
-            let groupID = MLSGroupID(welcomeBundle.id)
-            await uploadKeyPackagesIfNeeded()
-            staleKeyMaterialDetector.keyingMaterialUpdated(for: groupID)
-            return groupID
```
Comment (David-Henner) [inline wire-ios-data-model/Source/MLS/MLSService.swift:843]: are we completely removing this behaviour?

Comment (David-Henner) [reply wire-ios-data-model/Source/MLS/MLSService.swift:843]: nevermind

**Review by David-Henner — APPROVED** [review]

**Pull request merged**